src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -121,6 +124,171 @@ public BlittableObjectProperty(BlittableObjectInstance parent, string property)
+            }
+            
+            private unsafe bool TryGetValueFromCorax(BlittableObjectInstance parent, string property, IndexField indexField, out JsValue value)
+            {
+                value = null;
+                var fieldMapping = parent.IndexRetriever.KnownFields;
+                var isDynamic = indexField == null;
+                IndexFieldBinding binding = null;
+                if (isDynamic == false && fieldMapping.TryGetByFieldId(indexField.Id, out binding) == false)
+                    return false;
+
+                ref var reader = ref parent.IndexRetriever.CoraxEntry;
+                var fieldReader = binding is not null
+                    ? reader.GetFieldReaderFor(binding.Metadata)
+                    : reader.GetFieldReaderFor(parent.IndexRetriever.CoraxIndexSearcher.GetDynamicFieldName(property));
+
+                switch (fieldReader.Type)
+                {
+                    case IndexEntryFieldType.Empty:
+                        value = string.Empty;
+                        break;
+                    case IndexEntryFieldType.Null:
+                        value = DynamicJsNull.ExplicitNull;
+                        break;
+                    case IndexEntryFieldType.TupleListWithNulls:
+                    case IndexEntryFieldType.TupleList:
+                    {
+                        if (fieldReader.TryReadMany(out var iterator) == false)
+                            goto case IndexEntryFieldType.Invalid;
+
+                        var tupleList = new string[iterator.Count];
+
+                        for (int idX = 0; idX < iterator.Count && iterator.ReadNext(); ++idX)
+                        {
+                            if (iterator.IsNull)
+                                tupleList[idX] = null;
+                            else if (iterator.IsEmptyCollection)
+                                throw new InvalidDataException("Tuple list cannot contain an empty string (otherwise, where did the numeric came from!)");
+                            else
+                                tupleList[idX] = Encodings.Utf8.GetString(iterator.Sequence);
+                        }
+                        
+                        value = FromObject(parent.Engine, tupleList);
+                    }
+                        break;
+
+                    case IndexEntryFieldType.Tuple:
+                    {
+                        if (fieldReader.TryReadTuple(out long lVal, out double dVal, out Span<byte> valueInEntry) == false)
+                            goto case IndexEntryFieldType.Invalid;
+                        var hasTime = parent.IndexRetriever.IndexFieldsPersistence.HasTimeValues(binding?.FieldNameAsString ?? property);
+                        if (hasTime)
+                            value = Encodings.Utf8.GetString(valueInEntry);
+                        else if (valueInEntry.Contains((byte)'.'))
```

Comment:
Can we detect if that's long by checking if `dVal.AlmostEquals(lVal)`? If that's not the case then we know it's double